tools/check-magic-strings.sh

```diff
@@ -20,13 +20,17 @@ MAGIC_STRINGS_TEMP=`mktemp lit-magic-strings.inc.h.XXXXXXXXXX`
 
 cp $MAGIC_STRINGS_INC_H $MAGIC_STRINGS_TEMP
 $MAGIC_STRINGS_GEN
-diff -q $MAGIC_STRINGS_INC_H $MAGIC_STRINGS_TEMP
-DIFF_RESULT=$?
-mv $MAGIC_STRINGS_TEMP $MAGIC_STRINGS_INC_H
 
-if [ $DIFF_RESULT -ne 0 ]
-then
-  echo -e "\e[1;33m$MAGIC_STRINGS_INC_H must be re-generated. Run $MAGIC_STRINGS_GEN\e[0m"
+if [ $? -eq 0 ]; then
+  diff -q $MAGIC_STRINGS_INC_H $MAGIC_STRINGS_TEMP
+  DIFF_RESULT=$?
+  if [ $DIFF_RESULT -ne 0 ]
+  then
+    echo -e "\e[1;33m$MAGIC_STRINGS_INC_H must be re-generated. Run $MAGIC_STRINGS_GEN\e[0m"
+  fi
+else
+  DIFF_RESULT=$?
```

Comment:
Bash is a tricky beast. Here, `$?` is not the result of `$MAGIC_STRINGS_GEN` anymore but that of `[ $? -eq 0 ]`.

Try:
```sh
$ false; echo $?
1
$ false; if [ $? -eq 0 ]; then echo; else echo $?; fi
1
$ false; if [ $? -ne 0 ]; then echo $?; fi
0
```

So, it should be better to write:
```sh
$MAGIC_STRINGS_GEN
DIFF_RESULT=$?

if [ $DIFF_RESULT -eq 0 ]
then
  diff ...
  ...
fi
```